src/main/java/com/theironyard/controllers/IronGramController.java

```diff
@@ -0,0 +1,165 @@
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+
+@RestController
+public class IronGramController {
+    @Autowired
+    UserRepository users;
+
+    @Autowired
+    PhotoRepository photos;
+
+    Server dbui;
+
+    @PostConstruct
+    public void init() throws Exception {
+        dbui = Server.createWebServer().start();
+    }
+
+    @PreDestroy // stops web server before it crashes
+    public void destroy() {
+        dbui.stop();
+    }
+
+    @RequestMapping(path = "/login", method = RequestMethod.POST)
+    public User login(String userName, String password, HttpSession session, HttpServletResponse response) throws Exception {
+        User user = users.findFirstByName(userName);
+        if (user == null) {
+            user = new User(userName, PasswordStorage.createHash(password));
+            users.save(user);
+        } else if (!PasswordStorage.verifyPassword(password, user.getPassword())) {
+            throw new Exception("wrong password");
+        }
+        session.setAttribute("userName", userName);
+        response.sendRedirect("/");
+        return user;
+    }
+
+    @RequestMapping(path = "/logout")
+    public void logout(HttpSession session, HttpServletResponse response) throws IOException {
+        session.invalidate();
+        response.sendRedirect("/");
+    }
+
+
+    @RequestMapping(path = "/user", method = RequestMethod.GET)
+    public User getUser(HttpSession session) {
+        String userName = (String) session.getAttribute("userName");
+        return users.findFirstByName(userName);
+    }
+
+    @RequestMapping(path = "/upload", method = RequestMethod.POST)
+    public Photo upload(HttpSession session, HttpServletResponse response, String receiver, MultipartFile photo, boolean shared, String timer) throws Exception {
+        String userName = (String) session.getAttribute("userName");
+        if (userName == null) {
+            throw new Exception("Not logged in");
+        }
+        User senderUser = users.findFirstByName(userName);
+        User receiverUser = users.findFirstByName(receiver);
+        if (receiverUser == null) {
+            throw new Exception("receiver does not exist");
+        }
+//is it an image type file?
+        if (!photo.getContentType().startsWith("image")) {
+            throw new Exception("only images are allowed");
+        }
+// new file public: creates directory if it doesn't already have one
+        File photoFile = File.createTempFile("photo", photo.getOriginalFilename(), new File("public"));
+        FileOutputStream fos = new FileOutputStream(photoFile);
+        fos.write(photo.getBytes());
+
+        if (shared != true){
```

Comment:
Are you really doing anything here? If shared is true set to true? Huh?  Is this really what you meant?